src/Reports/Processor.ts

```diff
@@ -0,0 +1,196 @@
+      if (tranches.length === 0) throw new Error('No tranches found for snapshot')
+      return {
+        type: 'balanceSheet',
+        timestamp: snapshot.timestamp,
+        assetValuation: snapshot.portfolioValuation,
+        onchainReserve: snapshot.totalReserve,
+        offchainCash: snapshot.offchainCashValue,
+        accruedFees: snapshot.sumPoolFeesPendingAmount,
+        netAssetValue: snapshot.netAssetValue,
+        tranches: tranches?.map((tranche) => ({
+          name: tranche.pool.currency.symbol,
+          timestamp: tranche.timestamp,
+          tokenId: tranche.trancheId,
+          tokenSupply: tranche.tokenSupply,
+          tokenPrice: tranche.price,
+          trancheValue: tranche.tokenSupply.mul(tranche?.price?.toBigInt() ?? 0n),
+        })),
+        totalCapital: tranches.reduce(
+          (acc, curr) => acc.add(curr.tokenSupply.mul(curr?.price?.toBigInt() ?? 0n).toBigInt()),
+          new Currency(0, snapshot.poolCurrency.decimals)
+        ),
+      }
+    })
+    return this.applyGrouping<BalanceSheetReport>(items, filter?.groupBy, 'latest')
+  }
+
+  /**
+   * Process raw data into an aggregated cashflow report, fees and endCashBalance are NOT aggregated by period
+   * @param data Pool snapshot data
+   * @param filter Optional filtering and grouping options
+   * @returns Processed cashflow report at the end of each period
+   */
+  cashflow(data: CashflowData, filter?: ReportFilter): CashflowReport[] {
+    const subtype = data.metadata?.pool.asset.class === 'Public credit' ? 'publicCredit' : 'privateCredit'
+    const items: CashflowReport[] = data.poolSnapshots.map((day) => {
+      const poolFees =
+        data.poolFeeSnapshots[this.getDateKey(day.timestamp)]?.map((fee) => ({
+          name: fee.poolFee.name,
+          amount: fee.sumPaidAmountByPeriod,
+          timestamp: fee.timestamp,
+          feeId: fee.poolFeeId.split('-')[1] ?? '',
+        })) ?? []
+      const principalRepayments = day.sumPrincipalRepaidAmountByPeriod
+      const interest = day.sumInterestRepaidAmountByPeriod.add(day.sumUnscheduledRepaidAmountByPeriod)
+      const aquisistions = day.sumBorrowedAmountByPeriod
+      const fees = day.sumPoolFeesPaidAmountByPeriod
+      const netCashflowAsset = principalRepayments.sub(aquisistions).add(interest)
+      const investments = day.sumInvestedAmountByPeriod
+      const redemptions = day.sumRedeemedAmountByPeriod
+      const activitiesCashflow = investments.sub(redemptions)
+      const netCashflowAfterFees = netCashflowAsset.sub(fees)
+      const totalCashflow = netCashflowAfterFees.add(activitiesCashflow)
+      return {
+        type: 'cashflow',
+        subtype,
+        timestamp: day.timestamp,
+        principalPayments: principalRepayments,
+        ...(subtype === 'publicCredit' && { realizedPL: day.sumRealizedProfitFifoByPeriod }),
+        interestPayments: interest,
+        assetAcquisitions: aquisistions,
```

Comment:
Any specific reason to go with acquisitions here? Just asking because it is not a term we use anywhere else (either in the protocol code, subquery schema, or the UI).

Reply:
I was looking for a general term that I could use for "Asset purchases" and "Asset financings" so that I wouldn't have to make the distinction between public and private credit. But since the term isn't used anywhere else I guess it makes sense to leave them as separate keys